In [10]:
import pandas as pd

df = pd.read_csv("../data/all-states-history.csv")

In [11]:
df.head()

date state    death  deathConfirmed  deathIncrease  deathProbable  \
0  2021-03-07    AK    305.0             NaN              0            NaN   
1  2021-03-07    AL  10148.0          7963.0             -1         2185.0   
2  2021-03-07    AR   5319.0          4308.0             22         1011.0   
3  2021-03-07    AS      0.0             NaN              0            NaN   
4  2021-03-07    AZ  16328.0         14403.0              5         1925.0   

   hospitalized  hospitalizedCumulative  hospitalizedCurrently  \
0        1293.0                  1293.0                   33.0   
1       45976.0                 45976.0                  494.0   
2       14926.0                 14926.0                  335.0   
3           NaN                     NaN                    NaN   
4       57907.0                 57907.0                  963.0   

   hospitalizedIncrease  ...  totalTestResults  totalTestResultsIncrease  \
0                     0  ...         1731628.0                         0   
1                     0  ...         2323788.0                      2347   
2                    11  ...         2736442.0                      3380   
3                     0  ...            2140.0                         0   
4                    44  ...         7908105.0                     45110   

   totalTestsAntibody  totalTestsAntigen  totalTestsPeopleAntibody  \
0                 NaN                NaN                       NaN   
1                 NaN                NaN                  119757.0   
2                 NaN                NaN                       NaN   
3                 NaN                NaN                       NaN   
4            580569.0                NaN                  444089.0   

   totalTestsPeopleAntigen  totalTestsPeopleViral  \
0                      NaN                    NaN   
1                      NaN              2323788.0   
2                 481311.0                    NaN   
3                      NaN                    NaN   
4                      NaN              3842945.0   

   totalTestsPeopleViralIncrease  totalTestsViral  totalTestsViralIncrease  
0                              0        1731628.0                        0  
1                           2347              NaN                        0  
2                              0        2736442.0                     3380  
3                              0           2140.0                        0  
4                          14856        7908105.0                    45110  

[5 rows x 41 columns]

In [12]:
df.shape

(20780, 41)

Se analiza el tamaño del dataset para conocer el número de filas y columnas disponibles.El dataset contiene 20780 filas y 41 columnas.

In [13]:
df.info()


<class 'pandas.DataFrame'>
RangeIndex: 20780 entries, 0 to 20779
Data columns (total 41 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   date                              20780 non-null  str    
 1   state                             20780 non-null  str    
 2   death                             19930 non-null  float64
 3   deathConfirmed                    9422 non-null   float64
 4   deathIncrease                     20780 non-null  int64  
 5   deathProbable                     7593 non-null   float64
 6   hospitalized                      12382 non-null  float64
 7   hospitalizedCumulative            12382 non-null  float64
 8   hospitalizedCurrently             17339 non-null  float64
 9   hospitalizedIncrease              20780 non-null  int64  
 10  inIcuCumulative                   3789 non-null   float64
 11  inIcuCurrently                    11636 non-null  float64
 12  negative       

Se analiza la estructura del dataset para conocer el número de registros,
las variables disponibles y los tipos de datos.


In [14]:
df.describe()


death  deathConfirmed  deathIncrease  deathProbable  \
count  19930.000000     9422.000000   20780.000000    7593.000000   
mean    3682.216859     3770.182764      24.790712     417.291321   
std     6281.366321     4157.640633      60.162742     537.625982   
min        0.000000        0.000000    -201.000000       0.000000   
25%      161.250000      607.000000       0.000000      79.000000   
50%     1108.000000     2409.500000       6.000000     216.000000   
75%     4387.500000     5462.000000      24.000000     460.000000   
max    54124.000000    21177.000000    2559.000000    2594.000000   

       hospitalized  hospitalizedCumulative  hospitalizedCurrently  \
count  12382.000000            12382.000000           17339.000000   
mean    9262.762478             9262.762478            1190.576965   
std    12620.544081            12620.544081            2060.041207   
min        1.000000                1.000000               0.000000   
25%      985.250000              985.250000             166.500000   
50%     4472.000000             4472.000000             531.000000   
75%    12248.500000            12248.500000            1279.000000   
max    82237.000000            82237.000000           22851.000000   

       hospitalizedIncrease  inIcuCumulative  inIcuCurrently  ...  \
count          20780.000000      3789.000000    11636.000000  ...   
mean              37.360780      1934.191607      359.621176  ...   
std              208.237151      1953.329983      594.831150  ...   
min           -12257.000000         6.000000        0.000000  ...   
25%                0.000000       501.000000       60.000000  ...   
50%                0.000000      1295.000000      172.000000  ...   
75%               36.000000      2451.000000      380.000000  ...   
max            16373.000000      9263.000000     5225.000000  ...   

       totalTestResults  totalTestResultsIncrease  totalTestsAntibody  \
count      2.061400e+04              20780.000000        4.789000e+03   
mean       2.186936e+06              17508.388210        1.634028e+05   
std        4.436508e+06              33586.022461        2.016978e+05   
min        0.000000e+00            -130545.000000        0.000000e+00   
25%        1.040498e+05               1205.750000        1.896500e+04   
50%        6.552670e+05               6125.000000        8.465200e+04   
75%        2.264766e+06              19086.500000        2.300110e+05   
max        4.964601e+07             473076.000000        1.054711e+06   

       totalTestsAntigen  totalTestsPeopleAntibody  totalTestsPeopleAntigen  \
count       3.421000e+03               2200.000000               999.000000   
mean        3.089196e+05             165431.783182            168188.246246   
std         4.232854e+05             178181.836898            143748.073365   
min         1.000000e+00                  1.000000                 3.000000   
25%         2.004700e+04              54912.750000             37675.500000   
50%         1.233840e+05             103967.500000            144130.000000   
75%         4.327270e+05             183103.000000            255251.000000   
max         2.664340e+06             995580.000000            580372.000000   

       totalTestsPeopleViral  totalTestsPeopleViralIncrease  totalTestsViral  \
count           9.197000e+03                   2.078000e+04     1.451600e+04   
mean            9.650105e+05                   2.739554e+03     2.304555e+06   
std             1.415215e+06                   1.621900e+04     4.445365e+06   
min             0.000000e+00                  -1.043744e+06     0.000000e+00   
25%             1.414700e+05                   0.000000e+00     1.324600e+05   
50%             4.193720e+05                   0.000000e+00     7.316510e+05   
75%             1.229298e+06                   2.478000e+03     2.496925e+06   
max             1.124825e+07                   8.208170e+05     4.964601e+07   

       totalTestsViralIncrease  
count          

Se obtienen estadísticas descriptivas básicas de las variables numéricas
para comprender su distribución general.


In [15]:
df.columns


Index(['date', 'state', 'death', 'deathConfirmed', 'deathIncrease',
       'deathProbable', 'hospitalized', 'hospitalizedCumulative',
       'hospitalizedCurrently', 'hospitalizedIncrease', 'inIcuCumulative',
       'inIcuCurrently', 'negative', 'negativeIncrease',
       'negativeTestsAntibody', 'negativeTestsPeopleAntibody',
       'negativeTestsViral', 'onVentilatorCumulative', 'onVentilatorCurrently',
       'positive', 'positiveCasesViral', 'positiveIncrease', 'positiveScore',
       'positiveTestsAntibody', 'positiveTestsAntigen',
       'positiveTestsPeopleAntibody', 'positiveTestsPeopleAntigen',
       'positiveTestsViral', 'recovered', 'totalTestEncountersViral',
       'totalTestEncountersViralIncrease', 'totalTestResults',
       'totalTestResultsIncrease', 'totalTestsAntibody', 'totalTestsAntigen',
       'totalTestsPeopleAntibody', 'totalTestsPeopleAntigen',
       'totalTestsPeopleViral', 'totalTestsPeopleViralIncrease',
       'totalTestsViral', 'totalTestsViralIncrease'

Se listan los nombres de las columnas para identificar las variables
que serán utilizadas en el análisis.
